// @eog/geode-colors/index.js
var commonjsGlobal = typeof globalThis !== 'undefined' ? globalThis : typeof window !== 'undefined' ? window : typeof global !== 'undefined' ? global : typeof self !== 'undefined' ? self : {};

var check = function (it) {
  return it && it.Math == Math && it;
};

// https://github.com/zloirock/core-js/issues/86#issuecomment-115759028
var global$a =
  // eslint-disable-next-line es-x/no-global-this -- safe
  check(typeof globalThis == 'object' && globalThis) ||
  check(typeof window == 'object' && window) ||
  // eslint-disable-next-line no-restricted-globals -- safe
  check(typeof self == 'object' && self) ||
  check(typeof commonjsGlobal == 'object' && commonjsGlobal) ||
  // eslint-disable-next-line no-new-func -- fallback
  (function () { return this; })() || Function('return this')();

var objectGetOwnPropertyDescriptor = {};

var fails$a = function (exec) {
  try {
    return !!exec();
  } catch (error) {
    return true;
  }
};

var fails$9 = fails$a;

// Detect IE8's incomplete defineProperty implementation
var descriptors = !fails$9(function () {
  // eslint-disable-next-line es-x/no-object-defineproperty -- required for testing
  return Object.defineProperty({}, 1, { get: function () { return 7; } })[1] != 7;
});

var fails$8 = fails$a;

var functionBindNative = !fails$8(function () {
  // eslint-disable-next-line es-x/no-function-prototype-bind -- safe
  var test = (function () { /* empty */ }).bind();
  // eslint-disable-next-line no-prototype-builtins -- safe
  return typeof test != 'function' || test.hasOwnProperty('prototype');
});

var NATIVE_BIND$1 = functionBindNative;

var call$6 = Function.prototype.call;

var functionCall = NATIVE_BIND$1 ? call$6.bind(call$6) : function () {
  return call$6.apply(call$6, arguments);
};

var objectPropertyIsEnumerable = {};

var $propertyIsEnumerable = {}.propertyIsEnumerable;
// eslint-disable-next-line es-x/no-object-getownpropertydescriptor -- safe
var getOwnPropertyDescriptor$1 = Object.getOwnPropertyDescriptor;

// Nashorn ~ JDK8 bug
var NASHORN_BUG = getOwnPropertyDescriptor$1 && !$propertyIsEnumerable.call({ 1: 2 }, 1);

// `Object.prototype.propertyIsEnumerable` method implementation
// https://tc39.es/ecma262/#sec-object.prototype.propertyisenumerable
objectPropertyIsEnumerable.f = NASHORN_BUG ? function propertyIsEnumerable(V) {
  var descriptor = getOwnPropertyDescriptor$1(this, V);
  return !!descriptor && descriptor.enumerable;
} : $propertyIsEnumerable;

var createPropertyDescriptor$2 = function (bitmap, value) {
  return {
    enumerable: !(bitmap & 1),
    configurable: !(bitmap & 2),
    writable: !(bitmap & 4),
    value: value
  };
};

var NATIVE_BIND = functionBindNative;

var FunctionPrototype$1 = Function.prototype;
var bind = FunctionPrototype$1.bind;
var call$5 = FunctionPrototype$1.call;
var uncurryThis$b = NATIVE_BIND && bind.bind(call$5, call$5);

var functionUncurryThis = NATIVE_BIND ? function (fn) {
  return fn && uncurryThis$b(fn);
} : function (fn) {
  return fn && function () {
    return call$5.apply(fn, arguments);
  };
};

var uncurryThis$a = functionUncurryThis;

var toString$4 = uncurryThis$a({}.toString);
var stringSlice$1 = uncurryThis$a(''.slice);

var classofRaw$1 = function (it) {
  return stringSlice$1(toString$4(it), 8, -1);
};

var uncurryThis$9 = functionUncurryThis;
var fails$7 = fails$a;
var classof$2 = classofRaw$1;

var $Object$3 = Object;
var split = uncurryThis$9(''.split);

// fallback for non-array-like ES3 and non-enumerable old V8 strings
var indexedObject = fails$7(function () {
  // throws an error in rhino, see https://github.com/mozilla/rhino/issues/346
  // eslint-disable-next-line no-prototype-builtins -- safe
  return !$Object$3('z').propertyIsEnumerable(0);
}) ? function (it) {
  return classof$2(it) == 'String' ? split(it, '') : $Object$3(it);
} : $Object$3;

var $TypeError$5 = TypeError;

// `RequireObjectCoercible` abstract operation
// https://tc39.es/ecma262/#sec-requireobjectcoercible
var requireObjectCoercible$4 = function (it) {
  if (it == undefined) throw $TypeError$5("Can't call method on " + it);
  return it;
};

// toObject with fallback for non-array-like ES3 strings
var IndexedObject$1 = indexedObject;
var requireObjectCoercible$3 = requireObjectCoercible$4;

var toIndexedObject$3 = function (it) {
  return IndexedObject$1(requireObjectCoercible$3(it));
};

// `IsCallable` abstract operation
// https://tc39.es/ecma262/#sec-iscallable
var isCallable$b = function (argument) {
  return typeof argument == 'function';
};

var isCallable$a = isCallable$b;

var isObject$5 = function (it) {
  return typeof it == 'object' ? it !== null : isCallable$a(it);
};

var global$9 = global$a;
var isCallable$9 = isCallable$b;

var aFunction = function (argument) {
  return isCallable$9(argument) ? argument : undefined;
};

var getBuiltIn$3 = function (namespace, method) {
  return arguments.length < 2 ? aFunction(global$9[namespace]) : global$9[namespace] && global$9[namespace][method];
};

var uncurryThis$8 = functionUncurryThis;

var objectIsPrototypeOf = uncurryThis$8({}.isPrototypeOf);

var getBuiltIn$2 = getBuiltIn$3;

var engineUserAgent = getBuiltIn$2('navigator', 'userAgent') || '';

var global$8 = global$a;
var userAgent$1 = engineUserAgent;

var process = global$8.process;
var Deno = global$8.Deno;
var versions = process && process.versions || Deno && Deno.version;
var v8 = versions && versions.v8;
var match, version;

if (v8) {
  match = v8.split('.');
  // in old Chrome, versions of V8 isn't V8 = Chrome / 10
  // but their correct versions are not interesting for us
  version = match[0] > 0 && match[0] < 4 ? 1 : +(match[0] + match[1]);
}

// BrowserFS NodeJS `process` polyfill incorrectly set `.v8` to `0.0`
// so check `userAgent` even if `.v8` exists, but 0
if (!version && userAgent$1) {
  match = userAgent$1.match(/Edge\/(\d+)/);
  if (!match || match[1] >= 74) {
    match = userAgent$1.match(/Chrome\/(\d+)/);
    if (match) version = +match[1];
  }
}

var engineV8Version = version;

/* eslint-disable es-x/no-symbol -- required for testing */

var V8_VERSION = engineV8Version;
var fails$6 = fails$a;

// eslint-disable-next-line es-x/no-object-getownpropertysymbols -- required for testing
var nativeSymbol = !!Object.getOwnPropertySymbols && !fails$6(function () {
  var symbol = Symbol();
  // Chrome 38 Symbol has incorrect toString conversion
  // `get-own-property-symbols` polyfill symbols converted to object are not Symbol instances
  return !String(symbol) || !(Object(symbol) instanceof Symbol) ||
    // Chrome 38-40 symbols are not inherited from DOM collections prototypes to instances
    !Symbol.sham && V8_VERSION && V8_VERSION < 41;
});

/* eslint-disable es-x/no-symbol -- required for testing */

var NATIVE_SYMBOL$1 = nativeSymbol;

var useSymbolAsUid = NATIVE_SYMBOL$1
  && !Symbol.sham
  && typeof Symbol.iterator == 'symbol';

var getBuiltIn$1 = getBuiltIn$3;
var isCallable$8 = isCallable$b;
var isPrototypeOf$1 = objectIsPrototypeOf;
var USE_SYMBOL_AS_UID$1 = useSymbolAsUid;

var $Object$2 = Object;

var isSymbol$2 = USE_SYMBOL_AS_UID$1 ? function (it) {
  return typeof it == 'symbol';
} : function (it) {
  var $Symbol = getBuiltIn$1('Symbol');
  return isCallable$8($Symbol) && isPrototypeOf$1($Symbol.prototype, $Object$2(it));
};

var $String$2 = String;

var tryToString$1 = function (argument) {
  try {
    return $String$2(argument);
  } catch (error) {
    return 'Object';
  }
};

var isCallable$7 = isCallable$b;
var tryToString = tryToString$1;

var $TypeError$4 = TypeError;

// `Assert: IsCallable(argument) is true`
var aCallable$1 = function (argument) {
  if (isCallable$7(argument)) return argument;
  throw $TypeError$4(tryToString(argument) + ' is not a function');
};

var aCallable = aCallable$1;

// `GetMethod` abstract operation
// https://tc39.es/ecma262/#sec-getmethod
var getMethod$1 = function (V, P) {
  var func = V[P];
  return func == null ? undefined : aCallable(func);
};

var call$4 = functionCall;
var isCallable$6 = isCallable$b;
var isObject$4 = isObject$5;

var $TypeError$3 = TypeError;

// `OrdinaryToPrimitive` abstract operation
// https://tc39.es/ecma262/#sec-ordinarytoprimitive
var ordinaryToPrimitive$1 = function (input, pref) {
  var fn, val;
  if (pref === 'string' && isCallable$6(fn = input.toString) && !isObject$4(val = call$4(fn, input))) return val;
  if (isCallable$6(fn = input.valueOf) && !isObject$4(val = call$4(fn, input))) return val;
  if (pref !== 'string' && isCallable$6(fn = input.toString) && !isObject$4(val = call$4(fn, input))) return val;
  throw $TypeError$3("Can't convert object to primitive value");
};

var shared$3 = {exports: {}};

var global$7 = global$a;

// eslint-disable-next-line es-x/no-object-defineproperty -- safe
var defineProperty$2 = Object.defineProperty;

var defineGlobalProperty$3 = function (key, value) {
  try {
    defineProperty$2(global$7, key, { value: value, configurable: true, writable: true });
  } catch (error) {
    global$7[key] = value;
  } return value;
};

var global$6 = global$a;
var defineGlobalProperty$2 = defineGlobalProperty$3;

var SHARED = '__core-js_shared__';
var store$3 = global$6[SHARED] || defineGlobalProperty$2(SHARED, {});

var sharedStore = store$3;

var store$2 = sharedStore;

(shared$3.exports = function (key, value) {
  return store$2[key] || (store$2[key] = value !== undefined ? value : {});
})('versions', []).push({
  version: '3.24.1',
  mode: 'global',
  copyright: '© 2014-2022 Denis Pushkarev (zloirock.ru)',
  license: 'https://github.com/zloirock/core-js/blob/v3.24.1/LICENSE',
  source: 'https://github.com/zloirock/core-js'
});

var requireObjectCoercible$2 = requireObjectCoercible$4;

var $Object$1 = Object;

// `ToObject` abstract operation
// https://tc39.es/ecma262/#sec-toobject
var toObject$2 = function (argument) {
  return $Object$1(requireObjectCoercible$2(argument));
};

var uncurryThis$7 = functionUncurryThis;
var toObject$1 = toObject$2;

var hasOwnProperty = uncurryThis$7({}.hasOwnProperty);

// `HasOwnProperty` abstract operation
// https://tc39.es/ecma262/#sec-hasownproperty
// eslint-disable-next-line es-x/no-object-hasown -- safe
var hasOwnProperty_1 = Object.hasOwn || function hasOwn(it, key) {
  return hasOwnProperty(toObject$1(it), key);
};

var uncurryThis$6 = functionUncurryThis;

var id = 0;
var postfix = Math.random();
var toString$3 = uncurryThis$6(1.0.toString);

var uid$2 = function (key) {
  return 'Symbol(' + (key === undefined ? '' : key) + ')_' + toString$3(++id + postfix, 36);
};

var global$5 = global$a;
var shared$2 = shared$3.exports;
var hasOwn$7 = hasOwnProperty_1;
var uid$1 = uid$2;
var NATIVE_SYMBOL = nativeSymbol;
var USE_SYMBOL_AS_UID = useSymbolAsUid;

var WellKnownSymbolsStore = shared$2('wks');
var Symbol$1 = global$5.Symbol;
var symbolFor = Symbol$1 && Symbol$1['for'];
var createWellKnownSymbol = USE_SYMBOL_AS_UID ? Symbol$1 : Symbol$1 && Symbol$1.withoutSetter || uid$1;

var wellKnownSymbol$3 = function (name) {
  if (!hasOwn$7(WellKnownSymbolsStore, name) || !(NATIVE_SYMBOL || typeof WellKnownSymbolsStore[name] == 'string')) {
    var description = 'Symbol.' + name;
    if (NATIVE_SYMBOL && hasOwn$7(Symbol$1, name)) {
      WellKnownSymbolsStore[name] = Symbol$1[name];
    } else if (USE_SYMBOL_AS_UID && symbolFor) {
      WellKnownSymbolsStore[name] = symbolFor(description);
    } else {
      WellKnownSymbolsStore[name] = createWellKnownSymbol(description);
    }
  } return WellKnownSymbolsStore[name];
};

var call$3 = functionCall;
var isObject$3 = isObject$5;
var isSymbol$1 = isSymbol$2;
var getMethod = getMethod$1;
var ordinaryToPrimitive = ordinaryToPrimitive$1;
var wellKnownSymbol$2 = wellKnownSymbol$3;

var $TypeError$2 = TypeError;
var TO_PRIMITIVE = wellKnownSymbol$2('toPrimitive');

// `ToPrimitive` abstract operation
// https://tc39.es/ecma262/#sec-toprimitive
var toPrimitive$1 = function (input, pref) {
  if (!isObject$3(input) || isSymbol$1(input)) return input;
  var exoticToPrim = getMethod(input, TO_PRIMITIVE);
  var result;
  if (exoticToPrim) {
    if (pref === undefined) pref = 'default';
    result = call$3(exoticToPrim, input, pref);
    if (!isObject$3(result) || isSymbol$1(result)) return result;
    throw $TypeError$2("Can't convert object to primitive value");
  }
  if (pref === undefined) pref = 'number';
  return ordinaryToPrimitive(input, pref);
};

var toPrimitive = toPrimitive$1;
var isSymbol = isSymbol$2;

// `ToPropertyKey` abstract operation
// https://tc39.es/ecma262/#sec-topropertykey
var toPropertyKey$2 = function (argument) {
  var key = toPrimitive(argument, 'string');
  return isSymbol(key) ? key : key + '';
};

var global$4 = global$a;
var isObject$2 = isObject$5;

var document = global$4.document;
// typeof document.createElement is 'object' in old IE
var EXISTS$1 = isObject$2(document) && isObject$2(document.createElement);

var documentCreateElement = function (it) {
  return EXISTS$1 ? document.createElement(it) : {};
};

var DESCRIPTORS$7 = descriptors;
var fails$5 = fails$a;
var createElement = documentCreateElement;

// Thanks to IE8 for its funny defineProperty
var ie8DomDefine = !DESCRIPTORS$7 && !fails$5(function () {
  // eslint-disable-next-line es-x/no-object-defineproperty -- required for testing
  return Object.defineProperty(createElement('div'), 'a', {
    get: function () { return 7; }
  }).a != 7;
});

var DESCRIPTORS$6 = descriptors;
var call$2 = functionCall;
var propertyIsEnumerableModule$1 = objectPropertyIsEnumerable;
var createPropertyDescriptor$1 = createPropertyDescriptor$2;
var toIndexedObject$2 = toIndexedObject$3;
var toPropertyKey$1 = toPropertyKey$2;
var hasOwn$6 = hasOwnProperty_1;
var IE8_DOM_DEFINE$1 = ie8DomDefine;

// eslint-disable-next-line es-x/no-object-getownpropertydescriptor -- safe
var $getOwnPropertyDescriptor$1 = Object.getOwnPropertyDescriptor;

// `Object.getOwnPropertyDescriptor` method
// https://tc39.es/ecma262/#sec-object.getownpropertydescriptor
objectGetOwnPropertyDescriptor.f = DESCRIPTORS$6 ? $getOwnPropertyDescriptor$1 : function getOwnPropertyDescriptor(O, P) {
  O = toIndexedObject$2(O);
  P = toPropertyKey$1(P);
  if (IE8_DOM_DEFINE$1) try {
    return $getOwnPropertyDescriptor$1(O, P);
  } catch (error) { /* empty */ }
  if (hasOwn$6(O, P)) return createPropertyDescriptor$1(!call$2(propertyIsEnumerableModule$1.f, O, P), O[P]);
};

var objectDefineProperty = {};

var DESCRIPTORS$5 = descriptors;
var fails$4 = fails$a;

// V8 ~ Chrome 36-
// https://bugs.chromium.org/p/v8/issues/detail?id=3334
var v8PrototypeDefineBug = DESCRIPTORS$5 && fails$4(function () {
  // eslint-disable-next-line es-x/no-object-defineproperty -- required for testing
  return Object.defineProperty(function () { /* empty */ }, 'prototype', {
    value: 42,
    writable: false
  }).prototype != 42;
});

var isObject$1 = isObject$5;

var $String$1 = String;
var $TypeError$1 = TypeError;

// `Assert: Type(argument) is Object`
var anObject$4 = function (argument) {
  if (isObject$1(argument)) return argument;
  throw $TypeError$1($String$1(argument) + ' is not an object');
};

var DESCRIPTORS$4 = descriptors;
var IE8_DOM_DEFINE = ie8DomDefine;
var V8_PROTOTYPE_DEFINE_BUG = v8PrototypeDefineBug;
var anObject$3 = anObject$4;
var toPropertyKey = toPropertyKey$2;

var $TypeError = TypeError;
// eslint-disable-next-line es-x/no-object-defineproperty -- safe
var $defineProperty = Object.defineProperty;
// eslint-disable-next-line es-x/no-object-getownpropertydescriptor -- safe
var $getOwnPropertyDescriptor = Object.getOwnPropertyDescriptor;
var ENUMERABLE = 'enumerable';
var CONFIGURABLE$1 = 'configurable';
var WRITABLE = 'writable';

// `Object.defineProperty` method
// https://tc39.es/ecma262/#sec-object.defineproperty
objectDefineProperty.f = DESCRIPTORS$4 ? V8_PROTOTYPE_DEFINE_BUG ? function defineProperty(O, P, Attributes) {
  anObject$3(O);
  P = toPropertyKey(P);
  anObject$3(Attributes);
  if (typeof O === 'function' && P === 'prototype' && 'value' in Attributes && WRITABLE in Attributes && !Attributes[WRITABLE]) {
    var current = $getOwnPropertyDescriptor(O, P);
    if (current && current[WRITABLE]) {
      O[P] = Attributes.value;
      Attributes = {
        configurable: CONFIGURABLE$1 in Attributes ? Attributes[CONFIGURABLE$1] : current[CONFIGURABLE$1],
        enumerable: ENUMERABLE in Attributes ? Attributes[ENUMERABLE] : current[ENUMERABLE],
        writable: false
      };
    }
  } return $defineProperty(O, P, Attributes);
} : $defineProperty : function defineProperty(O, P, Attributes) {
  anObject$3(O);
  P = toPropertyKey(P);
  anObject$3(Attributes);
  if (IE8_DOM_DEFINE) try {
    return $defineProperty(O, P, Attributes);
  } catch (error) { /* empty */ }
  if ('get' in Attributes || 'set' in Attributes) throw $TypeError('Accessors not supported');
  if ('value' in Attributes) O[P] = Attributes.value;
  return O;
};

var DESCRIPTORS$3 = descriptors;
var definePropertyModule$2 = objectDefineProperty;
var createPropertyDescriptor = createPropertyDescriptor$2;

var createNonEnumerableProperty$2 = DESCRIPTORS$3 ? function (object, key, value) {
  return definePropertyModule$2.f(object, key, createPropertyDescriptor(1, value));
} : function (object, key, value) {
  object[key] = value;
  return object;
};

var makeBuiltIn$2 = {exports: {}};

var DESCRIPTORS$2 = descriptors;
var hasOwn$5 = hasOwnProperty_1;

var FunctionPrototype = Function.prototype;
// eslint-disable-next-line es-x/no-object-getownpropertydescriptor -- safe
var getDescriptor = DESCRIPTORS$2 && Object.getOwnPropertyDescriptor;

var EXISTS = hasOwn$5(FunctionPrototype, 'name');
// additional protection from minified / mangled / dropped function names
var PROPER = EXISTS && (function something() { /* empty */ }).name === 'something';
var CONFIGURABLE = EXISTS && (!DESCRIPTORS$2 || (DESCRIPTORS$2 && getDescriptor(FunctionPrototype, 'name').configurable));

var functionName = {
  EXISTS: EXISTS,
  PROPER: PROPER,
  CONFIGURABLE: CONFIGURABLE
};

var uncurryThis$5 = functionUncurryThis;
var isCallable$5 = isCallable$b;
var store$1 = sharedStore;

var functionToString = uncurryThis$5(Function.toString);

// this helper broken in `core-js@3.4.1-3.4.4`, so we can't use `shared` helper
if (!isCallable$5(store$1.inspectSource)) {
  store$1.inspectSource = function (it) {
    return functionToString(it);
  };
}

var inspectSource$2 = store$1.inspectSource;

var global$3 = global$a;
var isCallable$4 = isCallable$b;
var inspectSource$1 = inspectSource$2;

var WeakMap$1 = global$3.WeakMap;

var nativeWeakMap = isCallable$4(WeakMap$1) && /native code/.test(inspectSource$1(WeakMap$1));

var shared$1 = shared$3.exports;
var uid = uid$2;

var keys = shared$1('keys');

var sharedKey$1 = function (key) {
  return keys[key] || (keys[key] = uid(key));
};

var hiddenKeys$3 = {};

var NATIVE_WEAK_MAP = nativeWeakMap;
var global$2 = global$a;
var uncurryThis$4 = functionUncurryThis;
var isObject = isObject$5;
var createNonEnumerableProperty$1 = createNonEnumerableProperty$2;
var hasOwn$4 = hasOwnProperty_1;
var shared = sharedStore;
var sharedKey = sharedKey$1;
var hiddenKeys$2 = hiddenKeys$3;

var OBJECT_ALREADY_INITIALIZED = 'Object already initialized';
var TypeError$1 = global$2.TypeError;
var WeakMap = global$2.WeakMap;
var set, get, has;

var enforce = function (it) {
  return has(it) ? get(it) : set(it, {});
};

var getterFor = function (TYPE) {
  return function (it) {
    var state;
    if (!isObject(it) || (state = get(it)).type !== TYPE) {
      throw TypeError$1('Incompatible receiver, ' + TYPE + ' required');
    } return state;
  };
};

if (NATIVE_WEAK_MAP || shared.state) {
  var store = shared.state || (shared.state = new WeakMap());
  var wmget = uncurryThis$4(store.get);
  var wmhas = uncurryThis$4(store.has);
  var wmset = uncurryThis$4(store.set);
  set = function (it, metadata) {
    if (wmhas(store, it)) throw new TypeError$1(OBJECT_ALREADY_INITIALIZED);
    metadata.facade = it;
    wmset(store, it, metadata);
    return metadata;
  };
  get = function (it) {
    return wmget(store, it) || {};
  };
  has = function (it) {
    return wmhas(store, it);
  };
} else {
  var STATE = sharedKey('state');
  hiddenKeys$2[STATE] = true;
  set = function (it, metadata) {
    if (hasOwn$4(it, STATE)) throw new TypeError$1(OBJECT_ALREADY_INITIALIZED);
    metadata.facade = it;
    createNonEnumerableProperty$1(it, STATE, metadata);
    return metadata;
  };
  get = function (it) {
    return hasOwn$4(it, STATE) ? it[STATE] : {};
  };
  has = function (it) {
    return hasOwn$4(it, STATE);
  };
}

var internalState = {
  set: set,
  get: get,
  has: has,
  enforce: enforce,
  getterFor: getterFor
};

var fails$3 = fails$a;
var isCallable$3 = isCallable$b;
var hasOwn$3 = hasOwnProperty_1;
var DESCRIPTORS$1 = descriptors;
var CONFIGURABLE_FUNCTION_NAME = functionName.CONFIGURABLE;
var inspectSource = inspectSource$2;
var InternalStateModule = internalState;

var enforceInternalState = InternalStateModule.enforce;
var getInternalState = InternalStateModule.get;
// eslint-disable-next-line es-x/no-object-defineproperty -- safe
var defineProperty$1 = Object.defineProperty;

var CONFIGURABLE_LENGTH = DESCRIPTORS$1 && !fails$3(function () {
  return defineProperty$1(function () { /* empty */ }, 'length', { value: 8 }).length !== 8;
});

var TEMPLATE = String(String).split('String');

var makeBuiltIn$1 = makeBuiltIn$2.exports = function (value, name, options) {
  if (String(name).slice(0, 7) === 'Symbol(') {
    name = '[' + String(name).replace(/^Symbol\(([^)]*)\)/, '$1') + ']';
  }
  if (options && options.getter) name = 'get ' + name;
  if (options && options.setter) name = 'set ' + name;
  if (!hasOwn$3(value, 'name') || (CONFIGURABLE_FUNCTION_NAME && value.name !== name)) {
    if (DESCRIPTORS$1) defineProperty$1(value, 'name', { value: name, configurable: true });
    else value.name = name;
  }
  if (CONFIGURABLE_LENGTH && options && hasOwn$3(options, 'arity') && value.length !== options.arity) {
    defineProperty$1(value, 'length', { value: options.arity });
  }
  try {
    if (options && hasOwn$3(options, 'constructor') && options.constructor) {
      if (DESCRIPTORS$1) defineProperty$1(value, 'prototype', { writable: false });
    // in V8 ~ Chrome 53, prototypes of some methods, like `Array.prototype.values`, are non-writable
    } else if (value.prototype) value.prototype = undefined;
  } catch (error) { /* empty */ }
  var state = enforceInternalState(value);
  if (!hasOwn$3(state, 'source')) {
    state.source = TEMPLATE.join(typeof name == 'string' ? name : '');
  } return value;
};

// add fake Function#toString for correct work wrapped methods / constructors with methods like LoDash isNative
// eslint-disable-next-line no-extend-native -- required
Function.prototype.toString = makeBuiltIn$1(function toString() {
  return isCallable$3(this) && getInternalState(this).source || inspectSource(this);
}, 'toString');

var isCallable$2 = isCallable$b;
var definePropertyModule$1 = objectDefineProperty;
var makeBuiltIn = makeBuiltIn$2.exports;
var defineGlobalProperty$1 = defineGlobalProperty$3;

var defineBuiltIn$2 = function (O, key, value, options) {
  if (!options) options = {};
  var simple = options.enumerable;
  var name = options.name !== undefined ? options.name : key;
  if (isCallable$2(value)) makeBuiltIn(value, name, options);
  if (options.global) {
    if (simple) O[key] = value;
    else defineGlobalProperty$1(key, value);
  } else {
    try {
      if (!options.unsafe) delete O[key];
      else if (O[key]) simple = true;
    } catch (error) { /* empty */ }
    if (simple) O[key] = value;
    else definePropertyModule$1.f(O, key, {
      value: value,
      enumerable: false,
      configurable: !options.nonConfigurable,
      writable: !options.nonWritable
    });
  } return O;
};

var objectGetOwnPropertyNames = {};

var ceil$1 = Math.ceil;
var floor = Math.floor;

// `Math.trunc` method
// https://tc39.es/ecma262/#sec-math.trunc
// eslint-disable-next-line es-x/no-math-trunc -- safe
var mathTrunc = Math.trunc || function trunc(x) {
  var n = +x;
  return (n > 0 ? floor : ceil$1)(n);
};

var trunc = mathTrunc;

// `ToIntegerOrInfinity` abstract operation
// https://tc39.es/ecma262/#sec-tointegerorinfinity
var toIntegerOrInfinity$3 = function (argument) {
  var number = +argument;
  // eslint-disable-next-line no-self-compare -- NaN check
  return number !== number || number === 0 ? 0 : trunc(number);
};

var toIntegerOrInfinity$2 = toIntegerOrInfinity$3;

var max = Math.max;
var min$1 = Math.min;

// Helper for a popular repeating case of the spec:
// Let integer be ? ToInteger(index).
// If integer < 0, let result be max((length + integer), 0); else let result be min(integer, length).
var toAbsoluteIndex$1 = function (index, length) {
  var integer = toIntegerOrInfinity$2(index);
  return integer < 0 ? max(integer + length, 0) : min$1(integer, length);
};

var toIntegerOrInfinity$1 = toIntegerOrInfinity$3;

var min = Math.min;

// `ToLength` abstract operation
// https://tc39.es/ecma262/#sec-tolength
var toLength$2 = function (argument) {
  return argument > 0 ? min(toIntegerOrInfinity$1(argument), 0x1FFFFFFFFFFFFF) : 0; // 2 ** 53 - 1 == 9007199254740991
};

var toLength$1 = toLength$2;

// `LengthOfArrayLike` abstract operation
// https://tc39.es/ecma262/#sec-lengthofarraylike
var lengthOfArrayLike$1 = function (obj) {
  return toLength$1(obj.length);
};

var toIndexedObject$1 = toIndexedObject$3;
var toAbsoluteIndex = toAbsoluteIndex$1;
var lengthOfArrayLike = lengthOfArrayLike$1;

// `Array.prototype.{ indexOf, includes }` methods implementation
var createMethod$1 = function (IS_INCLUDES) {
  return function ($this, el, fromIndex) {
    var O = toIndexedObject$1($this);
    var length = lengthOfArrayLike(O);
    var index = toAbsoluteIndex(fromIndex, length);
    var value;
    // Array#includes uses SameValueZero equality algorithm
    // eslint-disable-next-line no-self-compare -- NaN check
    if (IS_INCLUDES && el != el) while (length > index) {
      value = O[index++];
      // eslint-disable-next-line no-self-compare -- NaN check
      if (value != value) return true;
    // Array#indexOf ignores holes, Array#includes - not
    } else for (;length > index; index++) {
      if ((IS_INCLUDES || index in O) && O[index] === el) return IS_INCLUDES || index || 0;
    } return !IS_INCLUDES && -1;
  };
};

var arrayIncludes = {
  // `Array.prototype.includes` method
  // https://tc39.es/ecma262/#sec-array.prototype.includes
  includes: createMethod$1(true),
  // `Array.prototype.indexOf` method
  // https://tc39.es/ecma262/#sec-array.prototype.indexof
  indexOf: createMethod$1(false)
};

var uncurryThis$3 = functionUncurryThis;
var hasOwn$2 = hasOwnProperty_1;
var toIndexedObject = toIndexedObject$3;
var indexOf = arrayIncludes.indexOf;
var hiddenKeys$1 = hiddenKeys$3;

var push = uncurryThis$3([].push);

var objectKeysInternal = function (object, names) {
  var O = toIndexedObject(object);
  var i = 0;
  var result = [];
  var key;
  for (key in O) !hasOwn$2(hiddenKeys$1, key) && hasOwn$2(O, key) && push(result, key);
  // Don't enum bug & hidden keys
  while (names.length > i) if (hasOwn$2(O, key = names[i++])) {
    ~indexOf(result, key) || push(result, key);
  }
  return result;
};

// IE8- don't enum bug keys
var enumBugKeys$2 = [
  'constructor',
  'hasOwnProperty',
  'isPrototypeOf',
  'propertyIsEnumerable',
  'toLocaleString',
  'toString',
  'valueOf'
];

var internalObjectKeys$1 = objectKeysInternal;
var enumBugKeys$1 = enumBugKeys$2;

var hiddenKeys = enumBugKeys$1.concat('length', 'prototype');

// `Object.getOwnPropertyNames` method
// https://tc39.es/ecma262/#sec-object.getownpropertynames
// eslint-disable-next-line es-x/no-object-getownpropertynames -- safe
objectGetOwnPropertyNames.f = Object.getOwnPropertyNames || function getOwnPropertyNames(O) {
  return internalObjectKeys$1(O, hiddenKeys);
};

var objectGetOwnPropertySymbols = {};

// eslint-disable-next-line es-x/no-object-getownpropertysymbols -- safe
objectGetOwnPropertySymbols.f = Object.getOwnPropertySymbols;

var getBuiltIn = getBuiltIn$3;
var uncurryThis$2 = functionUncurryThis;
var getOwnPropertyNamesModule = objectGetOwnPropertyNames;
var getOwnPropertySymbolsModule$1 = objectGetOwnPropertySymbols;
var anObject$2 = anObject$4;

var concat$1 = uncurryThis$2([].concat);

// all object keys, includes non-enumerable and symbols
var ownKeys$1 = getBuiltIn('Reflect', 'ownKeys') || function ownKeys(it) {
  var keys = getOwnPropertyNamesModule.f(anObject$2(it));
  var getOwnPropertySymbols = getOwnPropertySymbolsModule$1.f;
  return getOwnPropertySymbols ? concat$1(keys, getOwnPropertySymbols(it)) : keys;
};

var hasOwn$1 = hasOwnProperty_1;
var ownKeys = ownKeys$1;
var getOwnPropertyDescriptorModule = objectGetOwnPropertyDescriptor;
var definePropertyModule = objectDefineProperty;

var copyConstructorProperties$1 = function (target, source, exceptions) {
  var keys = ownKeys(source);
  var defineProperty = definePropertyModule.f;
  var getOwnPropertyDescriptor = getOwnPropertyDescriptorModule.f;
  for (var i = 0; i < keys.length; i++) {
    var key = keys[i];
    if (!hasOwn$1(target, key) && !(exceptions && hasOwn$1(exceptions, key))) {
      defineProperty(target, key, getOwnPropertyDescriptor(source, key));
    }
  }
};

var fails$2 = fails$a;
var isCallable$1 = isCallable$b;

var replacement = /#|\.prototype\./;

var isForced$1 = function (feature, detection) {
  var value = data[normalize(feature)];
  return value == POLYFILL ? true
    : value == NATIVE ? false
    : isCallable$1(detection) ? fails$2(detection)
    : !!detection;
};

var normalize = isForced$1.normalize = function (string) {
  return String(string).replace(replacement, '.').toLowerCase();
};

var data = isForced$1.data = {};
var NATIVE = isForced$1.NATIVE = 'N';
var POLYFILL = isForced$1.POLYFILL = 'P';

var isForced_1 = isForced$1;

var global$1 = global$a;
var getOwnPropertyDescriptor = objectGetOwnPropertyDescriptor.f;
var createNonEnumerableProperty = createNonEnumerableProperty$2;
var defineBuiltIn$1 = defineBuiltIn$2;
var defineGlobalProperty = defineGlobalProperty$3;
var copyConstructorProperties = copyConstructorProperties$1;
var isForced = isForced_1;

/*
  options.target         - name of the target object
  options.global         - target is the global object
  options.stat           - export as static methods of target
  options.proto          - export as prototype methods of target
  options.real           - real prototype method for the `pure` version
  options.forced         - export even if the native feature is available
  options.bind           - bind methods to the target, required for the `pure` version
  options.wrap           - wrap constructors to preventing global pollution, required for the `pure` version
  options.unsafe         - use the simple assignment of property instead of delete + defineProperty
  options.sham           - add a flag to not completely full polyfills
  options.enumerable     - export as enumerable property
  options.dontCallGetSet - prevent calling a getter on target
  options.name           - the .name of the function if it does not match the key
*/
var _export = function (options, source) {
  var TARGET = options.target;
  var GLOBAL = options.global;
  var STATIC = options.stat;
  var FORCED, target, key, targetProperty, sourceProperty, descriptor;
  if (GLOBAL) {
    target = global$1;
  } else if (STATIC) {
    target = global$1[TARGET] || defineGlobalProperty(TARGET, {});
  } else {
    target = (global$1[TARGET] || {}).prototype;
  }
  if (target) for (key in source) {
    sourceProperty = source[key];
    if (options.dontCallGetSet) {
      descriptor = getOwnPropertyDescriptor(target, key);
      targetProperty = descriptor && descriptor.value;
    } else targetProperty = target[key];
    FORCED = isForced(GLOBAL ? key : TARGET + (STATIC ? '.' : '#') + key, options.forced);
    // contained in target
    if (!FORCED && targetProperty !== undefined) {
      if (typeof sourceProperty == typeof targetProperty) continue;
      copyConstructorProperties(sourceProperty, targetProperty);
    }
    // add a flag to not completely full polyfills
    if (options.sham || (targetProperty && targetProperty.sham)) {
      createNonEnumerableProperty(sourceProperty, 'sham', true);
    }
    defineBuiltIn$1(target, key, sourceProperty, options);
  }
};

var internalObjectKeys = objectKeysInternal;
var enumBugKeys = enumBugKeys$2;

// `Object.keys` method
// https://tc39.es/ecma262/#sec-object.keys
// eslint-disable-next-line es-x/no-object-keys -- safe
var objectKeys$1 = Object.keys || function keys(O) {
  return internalObjectKeys(O, enumBugKeys);
};

var DESCRIPTORS = descriptors;
var uncurryThis$1 = functionUncurryThis;
var call$1 = functionCall;
var fails$1 = fails$a;
var objectKeys = objectKeys$1;
var getOwnPropertySymbolsModule = objectGetOwnPropertySymbols;
var propertyIsEnumerableModule = objectPropertyIsEnumerable;
var toObject = toObject$2;
var IndexedObject = indexedObject;

// eslint-disable-next-line es-x/no-object-assign -- safe
var $assign = Object.assign;
// eslint-disable-next-line es-x/no-object-defineproperty -- required for testing
var defineProperty = Object.defineProperty;
var concat = uncurryThis$1([].concat);

// `Object.assign` method
// https://tc39.es/ecma262/#sec-object.assign
var objectAssign = !$assign || fails$1(function () {
  // should have correct order of operations (Edge bug)
  if (DESCRIPTORS && $assign({ b: 1 }, $assign(defineProperty({}, 'a', {
    enumerable: true,
    get: function () {
      defineProperty(this, 'b', {
        value: 3,
        enumerable: false
      });
    }
  }), { b: 2 })).b !== 1) return true;
  // should work with symbols and should have deterministic property order (V8 bug)
  var A = {};
  var B = {};
  // eslint-disable-next-line es-x/no-symbol -- safe
  var symbol = Symbol();
  var alphabet = 'abcdefghijklmnopqrst';
  A[symbol] = 7;
  alphabet.split('').forEach(function (chr) { B[chr] = chr; });
  return $assign({}, A)[symbol] != 7 || objectKeys($assign({}, B)).join('') != alphabet;
}) ? function assign(target, source) { // eslint-disable-line no-unused-vars -- required for `.length`
  var T = toObject(target);
  var argumentsLength = arguments.length;
  var index = 1;
  var getOwnPropertySymbols = getOwnPropertySymbolsModule.f;
  var propertyIsEnumerable = propertyIsEnumerableModule.f;
  while (argumentsLength > index) {
    var S = IndexedObject(arguments[index++]);
    var keys = getOwnPropertySymbols ? concat(objectKeys(S), getOwnPropertySymbols(S)) : objectKeys(S);
    var length = keys.length;
    var j = 0;
    var key;
    while (length > j) {
      key = keys[j++];
      if (!DESCRIPTORS || call$1(propertyIsEnumerable, S, key)) T[key] = S[key];
    }
  } return T;
} : $assign;

var $$1 = _export;
var assign = objectAssign;

// `Object.assign` method
// https://tc39.es/ecma262/#sec-object.assign
// eslint-disable-next-line es-x/no-object-assign -- required for testing
$$1({ target: 'Object', stat: true, arity: 2, forced: Object.assign !== assign }, {
  assign: assign
});

var wellKnownSymbol$1 = wellKnownSymbol$3;

var TO_STRING_TAG$1 = wellKnownSymbol$1('toStringTag');
var test = {};

test[TO_STRING_TAG$1] = 'z';

var toStringTagSupport = String(test) === '[object z]';

var TO_STRING_TAG_SUPPORT = toStringTagSupport;
var isCallable = isCallable$b;
var classofRaw = classofRaw$1;
var wellKnownSymbol = wellKnownSymbol$3;

var TO_STRING_TAG = wellKnownSymbol('toStringTag');
var $Object = Object;

// ES3 wrong here
var CORRECT_ARGUMENTS = classofRaw(function () { return arguments; }()) == 'Arguments';

// fallback for IE11 Script Access Denied error
var tryGet = function (it, key) {
  try {
    return it[key];
  } catch (error) { /* empty */ }
};

// getting tag from ES6+ `Object.prototype.toString`
var classof$1 = TO_STRING_TAG_SUPPORT ? classofRaw : function (it) {
  var O, tag, result;
  return it === undefined ? 'Undefined' : it === null ? 'Null'
    // @@toStringTag case
    : typeof (tag = tryGet(O = $Object(it), TO_STRING_TAG)) == 'string' ? tag
    // builtinTag case
    : CORRECT_ARGUMENTS ? classofRaw(O)
    // ES3 arguments fallback
    : (result = classofRaw(O)) == 'Object' && isCallable(O.callee) ? 'Arguments' : result;
};

var classof = classof$1;

var $String = String;

var toString$2 = function (argument) {
  if (classof(argument) === 'Symbol') throw TypeError('Cannot convert a Symbol value to a string');
  return $String(argument);
};

var toIntegerOrInfinity = toIntegerOrInfinity$3;
var toString$1 = toString$2;
var requireObjectCoercible$1 = requireObjectCoercible$4;

var $RangeError = RangeError;

// `String.prototype.repeat` method implementation
// https://tc39.es/ecma262/#sec-string.prototype.repeat
var stringRepeat = function repeat(count) {
  var str = toString$1(requireObjectCoercible$1(this));
  var result = '';
  var n = toIntegerOrInfinity(count);
  if (n < 0 || n == Infinity) throw $RangeError('Wrong number of repetitions');
  for (;n > 0; (n >>>= 1) && (str += str)) if (n & 1) result += str;
  return result;
};

// https://github.com/tc39/proposal-string-pad-start-end
var uncurryThis = functionUncurryThis;
var toLength = toLength$2;
var toString = toString$2;
var $repeat = stringRepeat;
var requireObjectCoercible = requireObjectCoercible$4;

var repeat = uncurryThis($repeat);
var stringSlice = uncurryThis(''.slice);
var ceil = Math.ceil;

// `String.prototype.{ padStart, padEnd }` methods implementation
var createMethod = function (IS_END) {
  return function ($this, maxLength, fillString) {
    var S = toString(requireObjectCoercible($this));
    var intMaxLength = toLength(maxLength);
    var stringLength = S.length;
    var fillStr = fillString === undefined ? ' ' : toString(fillString);
    var fillLen, stringFiller;
    if (intMaxLength <= stringLength || fillStr == '') return S;
    fillLen = intMaxLength - stringLength;
    stringFiller = repeat(fillStr, ceil(fillLen / fillStr.length));
    if (stringFiller.length > fillLen) stringFiller = stringSlice(stringFiller, 0, fillLen);
    return IS_END ? S + stringFiller : stringFiller + S;
  };
};

var stringPad = {
  // `String.prototype.padStart` method
  // https://tc39.es/ecma262/#sec-string.prototype.padstart
  start: createMethod(false),
  // `String.prototype.padEnd` method
  // https://tc39.es/ecma262/#sec-string.prototype.padend
  end: createMethod(true)
};

// https://github.com/zloirock/core-js/issues/280
var userAgent = engineUserAgent;

var stringPadWebkitBug = /Version\/10(?:\.\d+){1,2}(?: [\w./]+)?(?: Mobile\/\w+)? Safari\//.test(userAgent);

var $ = _export;
var $padStart = stringPad.start;
var WEBKIT_BUG = stringPadWebkitBug;

// `String.prototype.padStart` method
// https://tc39.es/ecma262/#sec-string.prototype.padstart
$({ target: 'String', proto: true, forced: WEBKIT_BUG }, {
  padStart: function padStart(maxLength /* , fillString = ' ' */) {
    return $padStart(this, maxLength, arguments.length > 1 ? arguments[1] : undefined);
  }
});

var anObject$1 = anObject$4;

// `RegExp.prototype.flags` getter implementation
// https://tc39.es/ecma262/#sec-get-regexp.prototype.flags
var regexpFlags = function () {
  var that = anObject$1(this);
  var result = '';
  if (that.hasIndices) result += 'd';
  if (that.global) result += 'g';
  if (that.ignoreCase) result += 'i';
  if (that.multiline) result += 'm';
  if (that.dotAll) result += 's';
  if (that.unicode) result += 'u';
  if (that.unicodeSets) result += 'v';
  if (that.sticky) result += 'y';
  return result;
};

var call = functionCall;
var hasOwn = hasOwnProperty_1;
var isPrototypeOf = objectIsPrototypeOf;
var regExpFlags = regexpFlags;

var RegExpPrototype$1 = RegExp.prototype;

var regexpGetFlags = function (R) {
  var flags = R.flags;
  return flags === undefined && !('flags' in RegExpPrototype$1) && !hasOwn(R, 'flags') && isPrototypeOf(RegExpPrototype$1, R)
    ? call(regExpFlags, R) : flags;
};

var PROPER_FUNCTION_NAME = functionName.PROPER;
var defineBuiltIn = defineBuiltIn$2;
var anObject = anObject$4;
var $toString = toString$2;
var fails = fails$a;
var getRegExpFlags = regexpGetFlags;

var TO_STRING = 'toString';
var RegExpPrototype = RegExp.prototype;
var n$ToString = RegExpPrototype[TO_STRING];

var NOT_GENERIC = fails(function () { return n$ToString.call({ source: 'a', flags: 'b' }) != '/a/b'; });
// FF44- RegExp#toString has a wrong name
var INCORRECT_NAME = PROPER_FUNCTION_NAME && n$ToString.name != TO_STRING;

// `RegExp.prototype.toString` method
// https://tc39.es/ecma262/#sec-regexp.prototype.tostring
if (NOT_GENERIC || INCORRECT_NAME) {
  defineBuiltIn(RegExp.prototype, TO_STRING, function toString() {
    var R = anObject(this);
    var pattern = $toString(R.source);
    var flags = $toString(getRegExpFlags(R));
    return '/' + pattern + '/' + flags;
  }, { unsafe: true });
}

const setOpacity = (hex, alpha) => !hex ? hex : `${hex}${Math.floor(alpha * 255).toString(16).padStart(2)}`;

const color$v = {
  '0': '#EDFAFF',
  '05': '#C3EDFD',
  '5': '#C3EDFD',
  '10': '#8CDBF9',
  '20': '#58C6F2',
  '30': '#31B2EA',
  '40': '#139EDF',
  '50': '#0089D0',
  '60': '#0077BE',
  '70': '#0064A8',
  '80': '#004F8E',
  '90': '#003B71',
  '100': '#002954'
};

const color$u = {
  '0': '#EEFFF9',
  '05': '#C4FAE9',
  '5': '#C4FAE9',
  '10': '#94F1D4',
  '20': '#61DFB9',
  '30': '#3DC8A0',
  '40': '#22AB87',
  '50': '#0F8B6F',
  '60': '#056E5A',
  '70': '#005649',
  '80': '#00453E',
  '90': '#003A38',
  '100': '#002D2D'
};

const color$t = {
  '0': '#F3F8FF',
  '05': '#D9E7FD',
  '5': '#D9E7FD',
  '10': '#BAD3F9',
  '20': '#96BAF2',
  '30': '#77A2EA',
  '40': '#5B8BDF',
  '50': '#4475D0',
  '60': '#315FBE',
  '70': '#234CA8',
  '80': '#193A8E',
  '90': '#122A71',
  '100': '#0D1E54'
};

const color$s = {
  '0': '#F3F4F5',
  '05': '#EAEEF1',
  '5': '#EAEEF1',
  '10': '#D8DFE4',
  '20': '#CED7DE',
  '30': '#AAB8C2',
  '40': '#899BA8',
  '50': '#668093',
  '60': '#536878',
  '70': '#485867',
  '80': '#3D4956',
  '85': '#2E3842',
  '90': '#232D35',
  '95': '#191F26',
  '100': '#13171C'
};

const color$r = {
  '0': '#F4F4F4',
  '05': '#EEEEEE',
  '5': '#EEEEEE',
  '10': '#DEDFDF',
  '20': '#D6D7D8',
  '30': '#B6B8B9',
  '40': '#969A9C',
  '50': '#797D81',
  '60': '#636769',
  '70': '#56585B',
  '80': '#484A4D',
  '85': '#3B3C3D',
  '90': '#2B2C2E',
  '95': '#1E1F20',
  '100': '#161617'
};

const color$q = {
  '0': '#F1FFF1',
  '05': '#CCFBCB',
  '5': '#CCFBCB',
  '10': '#A1F29E',
  '20': '#74E170',
  '30': '#52CA4D',
  '40': '#38AF32',
  '50': '#26911E',
  '60': '#1B7511',
  '70': '#145F0A',
  '80': '#104E05',
  '90': '#0F4403',
  '100': '#0C3203'
};

const color$p = {
  '0': '#FEF4ED',
  '05': '#FEDDC3',
  '5': '#FEDDC3',
  '10': '#FBC190',
  '20': '#F6A05B',
  '30': '#EE8633',
  '40': '#E46E13',
  '50': '#D45A02',
  '60': '#C04E00',
  '70': '#A74100',
  '80': '#8B3300',
  '90': '#6C2600',
  '100': '#511700'
};

const color$o = {
  '0': '#FFF3F3',
  '05': '#FDD5D5',
  '5': '#FDD5D5',
  '10': '#FAB2B2',
  '20': '#F38A8A',
  '30': '#EB6868',
  '40': '#E04A4A',
  '50': '#D23131',
  '60': '#C01E1E',
  '70': '#AA1111',
  '80': '#910A0A',
  '90': '#750505',
  '100': '#590404'
};

const color$n = {
  '0': '#FFF4F0',
  '05': '#FDD8C8',
  '5': '#FDD8C8',
  '10': '#F9B69B',
  '20': '#F2936D',
  '30': '#EA774A',
  '40': '#DF5F2F',
  '50': '#D04A1A',
  '60': '#BE380B',
  '70': '#A82902',
  '80': '#8E1F00',
  '90': '#711700',
  '100': '#560200'
};

const color$m = {
  '0': '#FFF2F6',
  '05': '#FCD7E6',
  '5': '#FCD7E6',
  '10': '#F9B5D2',
  '20': '#F290BB',
  '30': '#E96FA8',
  '40': '#DD5297',
  '50': '#CE3A8A',
  '60': '#BB277E',
  '70': '#A41A74',
  '80': '#891167',
  '90': '#6B0B56',
  '100': '#4D0742'
};

const color$l = {
  '0': '#FBF6FF',
  '05': '#EFE0FD',
  '5': '#EFE0FD',
  '10': '#E1C5FA',
  '20': '#CDA7F3',
  '30': '#BB8CEB',
  '40': '#A673E1',
  '50': '#915DD3',
  '60': '#7A4BC2',
  '70': '#643CAD',
  '80': '#4F2F95',
  '90': '#3B257A',
  '100': '#2B1C5E'
};

const color$k = {
  '0': '#FEFFDA',
  '05': '#FEFFA3',
  '5': '#FEFFA3',
  '10': '#FFFF53',
  '20': '#FFEE00',
  '30': '#F1DD00',
  '40': '#DFC700',
  '50': '#C8AC00',
  '60': '#AD8D00',
  '70': '#916D00',
  '80': '#734F00',
  '90': '#553500',
  '100': '#482900'
};

const color$j = {
  '0': '#FEFFEE',
  '05': '#F6FBC0',
  '5': '#F6FBC0',
  '10': '#E8F28D',
  '20': '#D0E158',
  '30': '#B4CA32',
  '40': '#94AF16',
  '50': '#729105',
  '60': '#557500',
  '70': '#3F5F00',
  '80': '#2F4E00',
  '90': '#244400',
  '100': '#193401'
};

const color$i = {
  '0': '#FFF8E1',
  '05': '#FDE8AA',
  '5': '#FDE8AA',
  '10': '#FDD56B',
  '20': '#FBBB31',
  '30': '#EC9E06',
  '40': '#D98600',
  '50': '#C26E00',
  '60': '#AA5700',
  '70': '#8E4100',
  '80': '#773400',
  '90': '#672A00',
  '100': '#511E00'
};

const color$h = {
  "0": "#FFFFFF"
};

const color$g = {
  "100": "#000000"
};

var colors = /*#__PURE__*/Object.freeze({
	__proto__: null,
	blue: color$v,
	blueGreen: color$u,
	blueViolet: color$t,
	coolGray: color$s,
	gray: color$r,
	green: color$q,
	orange: color$p,
	red: color$o,
	redOrange: color$n,
	redViolet: color$m,
	violet: color$l,
	yellow: color$k,
	yellowGreen: color$j,
	yellowOrange: color$i,
	white: color$h,
	black: color$g
});

const color$f = {
  '0': 'color(display-p3 0.93 0.98 1)',
  '05': 'color(display-p3 0.76 0.93 0.99)',
  '5': 'color(display-p3 0.76 0.93 0.99)',
  '10': 'color(display-p3 0.55 0.86 0.98)',
  '20': 'color(display-p3 0.35 0.78 0.95)',
  '30': 'color(display-p3 0.19 0.7 0.92)',
  '40': 'color(display-p3 0.93 0.98 1)',
  '50': 'color(display-p3 0 0.54 0.82)',
  '60': 'color(display-p3 0 0.47 0.75)',
  '70': 'color(display-p3 0 0.39 0.66)',
  '80': 'color(display-p3 0 0.31 0.56)',
  '90': 'color(display-p3 0 0.23 0.44)',
  '100': 'color(display-p3 0 0.16 0.33)'
};

const color$e = {
  '0': 'color(display-p3 0.93 1 0.98)',
  '05': 'color(display-p3 0.77 0.98 0.91)',
  '5': 'color(display-p3 0.77 0.98 0.91)',
  '10': 'color(display-p3 0.58 0.95 0.83)',
  '20': 'color(display-p3 0.38 0.87 0.72)',
  '30': 'color(display-p3 0.24 0.78 0.63)',
  '40': 'color(display-p3 0.13 0.67 0.53)',
  '50': 'color(display-p3 0.06 0.55 0.44)',
  '60': 'color(display-p3 0.02 0.43 0.35)',
  '70': 'color(display-p3 0 0.34 0.29)',
  '80': 'color(display-p3 0 0.27 0.25)',
  '90': 'color(display-p3 0 0.23 0.22)',
  '100': 'color(display-p3 0 0.18)'
};

const color$d = {
  '0': 'color(display-p3 0.95 0.97 1)',
  '05': 'color(display-p3 0.85 0.91 0.99)',
  '5': 'color(display-p3 0.85 0.91 0.99)',
  '10': 'color(display-p3 0.73 0.83 0.98)',
  '20': 'color(display-p3 0.59 0.73 0.95)',
  '30': 'color(display-p3 0.47 0.64 0.92)',
  '40': 'color(display-p3 0.36 0.55 0.87)',
  '50': 'color(display-p3 0.27 0.46 0.82)',
  '60': 'color(display-p3 0.19 0.37 0.75)',
  '70': 'color(display-p3 0.14 0.3 0.66)',
  '80': 'color(display-p3 0.1 0.23 0.56)',
  '90': 'color(display-p3 0.07 0.16 0.44)',
  '100': 'color(display-p3 0.05 0.12 0.33)'
};

const color$c = {
  '0': 'color(display-p3 0.95 0.96 0.96)',
  '05': 'color(display-p3 0.92 0.93 0.95)',
  '5': 'color(display-p3 0.92 0.93 0.95)',
  '10': 'color(display-p3 0.85 0.87 0.89)',
  '20': 'color(display-p3 0.91 0.84 0.87)',
  '30': 'color(display-p3 0.67 0.72 0.76)',
  '40': 'color(display-p3 0.54 0.61 0.66)',
  '50': 'color(display-p3 0.4 0.5 0.58)',
  '60': 'color(display-p3 0.33 0.41 0.47)',
  '70': 'color(display-p3 0.28 0.35 0.4)',
  '80': 'color(display-p3 0.24 0.29 0.34)',
  '85': 'color(display-p3 0.18 0.22 0.26)',
  '90': 'color(display-p3 0.14 0.18 0.21)',
  '95': 'color(display-p3 0.1 0.12 0.15)',
  '100': 'color(display-p3 0.07 0.09 0.11)'
};

const color$b = {
  '0': 'color(display-p3 0.96 0.96 0.96)',
  '05': 'color(display-p3 0.93 0.93 0.93)',
  '5': 'color(display-p3 0.93 0.93 0.93)',
  '10': 'color(display-p3 0.87 0.87 0.97)',
  '20': 'color(display-p3 0.84 0.84 0.85)',
  '30': 'color(display-p3 0.71 0.72 0.73)',
  '40': 'color(display-p3 0.59 0.6 0.61)',
  '50': 'color(display-p3 0.47 0.49 0.51)',
  '60': 'color(display-p3 0.39 0.4 0.41)',
  '70': 'color(display-p3 0.34 0.35 0.36)',
  '80': 'color(display-p3 0.28 0.29 0.3)',
  '85': 'color(display-p3 0.23 0.24 0.24)',
  '90': 'color(display-p3 0.170.17 0.18)',
  '95': 'color(display-p3 0.12 0.12 0.13)',
  '100': 'color(display-p3 0.09 0.09 0.09)'
};

const color$a = {
  '0': 'color(display-p3 0.95 1 0.95)',
  '05': 'color(display-p3 0.8 0.98 0.8)',
  '5': 'color(display-p3 0.8 0.98 0.8)',
  '10': 'color(display-p3 0.63 0.95 0.62)',
  '20': 'color(display-p3 0.45 0.88 0.44)',
  '30': 'color(display-p3 0.32 0.79 0.3)',
  '40': 'color(display-p3 0.22 0.69 0.2)',
  '50': 'color(display-p3 0.15 0.57 0.12)',
  '60': 'color(display-p3 0.11 0.46 0.07)',
  '70': 'color(display-p3 0.08 0.37 0.04)',
  '80': 'color(display-p3 0.06 0.31 0.02)',
  '90': 'color(display-p3 0.06 0.27 0.01)',
  '100': 'color(display-p3 0.05 0.2 0.01)'
};

const color$9 = {
  '0': 'color(display-p3 1 0.96 0.93)',
  '05': 'color(display-p3 1 0.87 0.76)',
  '5': 'color(display-p3 1 0.87 0.76)',
  '10': 'color(display-p3 0.98 0.76 0.56)',
  '20': 'color(display-p3 0.96 0.63 0.36)',
  '30': 'color(display-p3 0.93 0.53 0.2)',
  '40': 'color(display-p3 0.89 0.43 0.07)',
  '50': 'color(display-p3 0.83 0.35 0.01)',
  '60': 'color(display-p3 0.75 0.31 0)',
  '70': 'color(display-p3 0.65 0.25 0)',
  '80': 'color(display-p3 0.55 0.2 0)',
  '90': 'color(display-p3 0.42 0.15 0)',
  '100': 'color(display-p3 0.32 0.09 0)'
};

const color$8 = {
  '0': 'color(display-p3 1 0.95 0.95)',
  '05': 'color(display-p3 0.99 0.84 0.84)',
  '5': 'color(display-p3 0.99 0.84 0.84)',
  '10': 'color(display-p3 0.98 0.7 0.7)',
  '20': 'color(display-p3 0.95 0.54 0.54)',
  '30': 'color(display-p3 0.92 0.41 0.41)',
  '40': 'color(display-p3 0.88 0.29 0.29)',
  '50': 'color(display-p3 0.82 0.19 0.19)',
  '60': 'color(display-p3 0.75 0.12 0.12)',
  '70': 'color(display-p3 0.67 0.07 0.07)',
  '80': 'color(display-p3 0.54 0.04 0.04)',
  '90': 'color(display-p3 0.46 0.02 0.02)',
  '100': 'color(display-p3 0.35 0.02 0.02)'
};

const color$7 = {
  '0': 'color(display-p3 1 .96 .94)',
  '05': 'color(display-p3 0.99 0.85 0.78)',
  '5': 'color(display-p3 0.99 0.85 0.78)',
  '10': 'color(display-p3 0.98 0.71 0.61)',
  '20': 'color(display-p3 0.95 0.58 0.43)',
  '30': 'color(display-p3 0.92 0.47 0.29)',
  '40': 'color(display-p3 0.87 0.37 0.18)',
  '50': 'color(display-p3 0.82 0.29 0.1)',
  '60': 'color(display-p3 0.75 0.22 0.04)',
  '70': 'color(display-p3 0.66 0.16 0.01)',
  '80': 'color(display-p3 0.56 0.12 0)',
  '90': 'color(display-p3 0.44 0.09 0)',
  '100': 'color(display-p3 0.34 0.01 0)'
};

const color$6 = {
  '0': 'color(display-p3 1 0.95 0.96)',
  '05': 'color(display-p3 0.99 0.84 0.9)',
  '5': 'color(display-p3 0.99 0.84 0.9)',
  '10': 'color(display-p3 0.98 0.71 0.82)',
  '20': 'color(display-p3 0.95 0.56 0.73)',
  '30': 'color(display-p3 0.91 0.44 0.66)',
  '40': 'color(display-p3 0.87 0.32 0.59)',
  '50': 'color(display-p3 0.81 0.23 0.54)',
  '60': 'color(display-p3 0.73 0.15 0.49)',
  '70': 'color(display-p3 0.64 0.1 0.45)',
  '80': 'color(display-p3 0.54 0.07 0.4)',
  '90': 'color(display-p3 0.42 0.04 0.34)',
  '100': 'color(display-p3 0.3 0.03 0.26)'
};

const color$5 = {
  '0': 'color(display-p3 0.98 0.96 1)',
  '05': 'color(display-p3 0.94 0.88 0.99)',
  '5': 'color(display-p3 0.94 0.88 0.99)',
  '10': 'color(display-p3 0.88 0.77 0.98)',
  '20': 'color(display-p3 0.8 0.65 0.95)',
  '30': 'color(display-p3 0.73 0.55 0.92)',
  '40': 'color(display-p3 0.65 0.45 0.88)',
  '50': 'color(display-p3 0.57 0.36 0.83)',
  '60': 'color(display-p3 0.48 0.29 0.76)',
  '70': 'color(display-p3 0.39 0.24 0.68)',
  '80': 'color(display-p3 0.21 0.18 0.58)',
  '90': 'color(display-p3 0.23 0.15 0.48)',
  '100': 'color(display-p3 0.17 0.11 0.37)'
};

const color$4 = {
  '0': 'color(display-p3 1 1 0.85)',
  '05': 'color(display-p3 0.99 1 0.64)',
  '5': 'color(display-p3 0.99 1 0.64)',
  '10': 'color(display-p3 1 1 0.33)',
  '20': 'color(display-p3 1 0.93 0)',
  '30': 'color(display-p3 0.95 0.87 0)',
  '40': 'color(display-p3 0.87 0.87 0)',
  '50': 'color(display-p3 0.78 0.67 0)',
  '60': 'color(display-p3 0.68 0.55 0)',
  '70': 'color(display-p3 0.57 0.43 0)',
  '80': 'color(display-p3 0.45 0.31 0)',
  '90': 'color(display-p3 0.33 0.21 0)',
  '100': 'color(display-p3 0.28 0.16 0)'
};

const color$3 = {
  '0': 'color(display-p3 1 1 0.93)',
  '05': 'color(display-p3 0.96 0.98 0.75)',
  '5': 'color(display-p3 0.96 0.98 0.75)',
  '10': 'color(display-p3 0.91 0.95 0.55)',
  '20': 'color(display-p3 0.82 0.88 0.35)',
  '30': 'color(display-p3 0.71 0.79 0.2)',
  '40': 'color(display-p3 0.58 0.69 0.09)',
  '50': 'color(display-p3 0.45 0.57 0.02)',
  '60': 'color(display-p3 0.33 0.46 0)',
  '70': 'color(display-p3 0.25 0.37 0)',
  '80': 'color(display-p3 0.18 0.31 0)',
  '90': 'color(display-p3 0.14 0.27 0)',
  '100': 'color(display-p3 0.1 0.2 0)'
};

const color$2 = {
  '0': 'color(display-p3 1 0.97 0.88)',
  '05': 'color(display-p3 0.99 0.91 0.67)',
  '5': 'color(display-p3 0.99 0.91 0.67)',
  '10': 'color(display-p3 0.99 0.84 0.42)',
  '20': 'color(display-p3 0.98 0.73 0.19)',
  '30': 'color(display-p3 0.93 0.62 0.02)',
  '40': 'color(display-p3 0.85 0.53 0)',
  '50': 'color(display-p3 0.76 0.43 0)',
  '60': 'color(display-p3 0.67 0.34 0)',
  '70': 'color(display-p3 0.56 0.25 0)',
  '80': 'color(display-p3 0.47 0.2 0)',
  '90': 'color(display-p3 0.4 0.16 0)',
  '100': 'color(display-p3 0.32 0.12 0)'
};

const color$1 = {
  "0": "color(display-p3 1 1 1 1)"
};

const color = {
  "100": "color(display-p3 0 0 0 1)"
};

var p3Colors = /*#__PURE__*/Object.freeze({
	__proto__: null,
	blue: color$f,
	blueGreen: color$e,
	blueViolet: color$d,
	coolGray: color$c,
	gray: color$b,
	green: color$a,
	orange: color$9,
	red: color$8,
	redOrange: color$7,
	redViolet: color$6,
	violet: color$5,
	yellow: color$4,
	yellowGreen: color$3,
	yellowOrange: color$2,
	white: color$1,
	black: color
});

const geodeColors$1 = Object.assign({}, colors);
const p3geodeColors = Object.assign({}, p3Colors);
const chartColors = {
  ruby: geodeColors$1.red[50],
  amber: geodeColors$1.orange[40],
  citrine: geodeColors$1.yellowOrange[20],
  emerald: geodeColors$1.green[30],
  amazonite: geodeColors$1.blueGreen[30],
  apatite: geodeColors$1.blue[20],
  sapphire: geodeColors$1.blue[40],
  amethyst: geodeColors$1.violet[70],
  iolite: geodeColors$1.violet[90],
  graphite: geodeColors$1.gray[50],
  obsidian: geodeColors$1.black[100],
  calcite: 'rgba(250, 250, 250, 1)',
  streams: {
    streamBoe: geodeColors$1.black[100],
    streamGas: geodeColors$1.red[40],
    streamNgl: geodeColors$1.blue[40],
    streamOil: geodeColors$1.green[30],
    streamWater: geodeColors$1.blue[20]
  }
};
const lightPalette = {
  action: {
    activatedOpacity: 0.24,
    active: 'rgba(0, 0, 0, 0.9)',
    disabled: 'rgba(0, 0, 0, 0.26)',
    disabledBackground: 'rgba(0, 0, 0, 0.12)',
    disabledOpacity: 0.38,
    focus: 'rgba(0, 0, 0, 0.12)',
    focusOpacity: 0.12,
    hover: 'rgba(0, 0, 0, 0.04)',
    hoverOpacity: 0.04,
    selected: 'rgba(0, 0, 0, 0.08)',
    selectedOpacity: 0.08
  },
  background: {
    default: '#F8F9FA',
    paper: geodeColors$1.white[0]
  },
  borders: {
    divider: '1px solid rgba(19, 23, 28, 0.1)',
    outline: '1px solid rgb(224, 224, 224)'
  },
  chartColors,
  coloredBackgrounds: {
    customUI: 'rgba(170, 184, 194, 0.08)',
    error: geodeColors$1.red[0],
    info: geodeColors$1.blueViolet[5],
    primary: 'rgba(88, 198, 242, 0.23)',
    secondary: 'rgba(205, 167, 243, 0.08)',
    success: geodeColors$1.green[0],
    warning: geodeColors$1.yellowOrange[0]
  },
  coloredText: {
    error: geodeColors$1.red[90],
    info: geodeColors$1.blueViolet[90],
    success: geodeColors$1.green[70],
    warning: geodeColors$1.yellowOrange[90]
  },
  primary: {
    main: geodeColors$1.blue[40],
    dark: geodeColors$1.blue[70],
    light: geodeColors$1.blue[5],
    background: geodeColors$1.blue[0],
    contrastText: geodeColors$1.white[0],

    get hover() {
      return setOpacity(this.main, 0.1);
    },

    get selected() {
      return setOpacity(this.main, 0.2);
    },

    get focused() {
      return setOpacity(this.main, 0.25);
    }

  },
  customUI: {
    main: geodeColors$1.coolGray[50],
    dark: geodeColors$1.coolGray[70],
    light: geodeColors$1.coolGray[10],
    background: geodeColors$1.coolGray[0],
    contrastText: geodeColors$1.white[0],

    get hover() {
      return setOpacity(this.main, 0.1);
    },

    get selected() {
      return setOpacity(this.main, 0.2);
    },

    get focused() {
      return setOpacity(this.main, 0.25);
    }

  },
  info: {
    main: geodeColors$1.blueViolet[50],
    dark: geodeColors$1.blueViolet[70],
    light: geodeColors$1.blueViolet[5],
    background: geodeColors$1.blueViolet[0],
    contrastText: geodeColors$1.white[0],

    get hover() {
      return setOpacity(this.main, 0.1);
    },

    get selected() {
      return setOpacity(this.main, 0.2);
    },

    get focused() {
      return setOpacity(this.main, 0.25);
    }

  },
  success: {
    main: geodeColors$1.green[40],
    dark: geodeColors$1.green[60],
    light: geodeColors$1.green[5],
    background: geodeColors$1.green[0],
    contrastText: geodeColors$1.white[0],

    get hover() {
      return setOpacity(this.main, 0.1);
    },

    get selected() {
      return setOpacity(this.main, 0.2);
    },

    get focused() {
      return setOpacity(this.main, 0.25);
    }

  },
  warning: {
    main: geodeColors$1.yellowOrange[30],
    dark: geodeColors$1.yellowOrange[60],
    light: geodeColors$1.yellowOrange[5],
    background: geodeColors$1.yellowOrange[0],
    contrastText: geodeColors$1.black[100],

    get hover() {
      return setOpacity(this.main, 0.1);
    },

    get selected() {
      return setOpacity(this.main, 0.2);
    },

    get focused() {
      return setOpacity(this.main, 0.25);
    }

  },
  error: {
    main: geodeColors$1.red[50],
    dark: geodeColors$1.red[70],
    light: geodeColors$1.red[5],
    background: geodeColors$1.red[0],
    contrastText: geodeColors$1.white[0],

    get hover() {
      return setOpacity(this.main, 0.1);
    },

    get selected() {
      return setOpacity(this.main, 0.2);
    },

    get focused() {
      return setOpacity(this.main, 0.25);
    }

  },
  miscellaneous: {
    backdrop: 'rgba(19, 23, 28, 0.5)',
    divider: 'rgba(19, 23, 28, 0.1)',
    filledInput: 'rgba(83, 104, 120, 0.09)',
    ratingFull: geodeColors$1.yellowOrange[20],
    ratingUnfilled: '#FBBB31',
    snackBar: geodeColors$1.coolGray[90],
    tooltip: 'rgba(35, 45, 53, 0.9)'
  },
  secondary: {
    contrastText: geodeColors$1.white[0],
    dark: geodeColors$1.violet[70],
    light: geodeColors$1.violet[10],
    main: geodeColors$1.violet[50],

    get hover() {
      return setOpacity(this.main, 0.1);
    },

    get selected() {
      return setOpacity(this.main, 0.2);
    },

    get focused() {
      return setOpacity(this.main, 0.25);
    }

  },
  text: {
    disabled: 'rgba(19, 23, 28, 0.25)',
    hint: 'rgba(19, 23, 28, 0.35)',
    primary: 'rgba(19, 23, 28, 0.9)',
    secondary: 'rgba(19, 23, 28, 0.54)'
  },
  stroke: {
    divider: 'rgba(19, 23, 28, 0.10)',
    stroke: 'rgba(19, 23, 28, 0.20)',
    hover: 'rgba(19, 23, 28, 0.50)',
    disabled: 'rgba(19, 23, 28, 0.10)'
  },
  type: 'light'
};
const darkPalette = {
  action: {
    activatedOpacity: 0.24,
    active: 'rgba(255, 255, 255, 0.9)',
    disabled: 'rgba(255, 255, 255, 0.3)',
    disabledBackground: 'rgba(255, 255, 255, 0.12)',
    disabledOpacity: 0.38,
    focus: 'rgba(255, 255, 255, 0.12)',
    focusOpacity: 0.12,
    hover: 'rgba(255, 255, 255, 0.08)',
    hoverOpacity: 0.08,
    selected: 'rgba(255, 255, 255, 0.16)',
    selectedOpacity: 0.16
  },
  background: {
    default: geodeColors$1.coolGray[100],
    paper: geodeColors$1.coolGray[95]
  },
  borders: {
    divider: '1px solid rgba(137, 155, 168, 0.12)',
    outline: '1px solid rgb(224, 224, 224)'
  },
  chartColors,
  coloredBackgrounds: {
    customUI: 'rgba(61, 73, 86, 0.3)',
    error: geodeColors$1.red[5],
    info: geodeColors$1.blueViolet[5],
    primary: 'rgba(0, 79, 142, 0.2)',
    secondary: 'rgba(205, 167, 243, 0.2)',
    success: geodeColors$1.green[5],
    warning: geodeColors$1.yellowOrange[5]
  },
  coloredText: {
    error: geodeColors$1.red[90],
    info: geodeColors$1.blueViolet[90],
    success: geodeColors$1.green[70],
    warning: geodeColors$1.yellowOrange[90]
  },
  primary: {
    main: geodeColors$1.blue[40],
    dark: geodeColors$1.blue[90],
    light: geodeColors$1.blue[20],
    background: geodeColors$1.blue[100],
    contrastText: geodeColors$1.white[0],

    get hover() {
      return setOpacity(this.main, 0.1);
    },

    get selected() {
      return setOpacity(this.main, 0.2);
    },

    get focused() {
      return setOpacity(this.main, 0.25);
    }

  },
  customUI: {
    main: geodeColors$1.coolGray[40],
    dark: geodeColors$1.coolGray[80],
    light: geodeColors$1.coolGray[20],
    background: geodeColors$1.coolGray[90],
    contrastText: geodeColors$1.black[100],

    get hover() {
      return setOpacity(this.main, 0.1);
    },

    get selected() {
      return setOpacity(this.main, 0.2);
    },

    get focused() {
      return setOpacity(this.main, 0.25);
    }

  },
  info: {
    main: geodeColors$1.blueViolet[40],
    dark: geodeColors$1.blueViolet[90],
    light: geodeColors$1.blueViolet[10],
    background: geodeColors$1.blueViolet[100],
    contrastText: geodeColors$1.white[0],

    get hover() {
      return setOpacity(this.main, 0.15);
    },

    get selected() {
      return setOpacity(this.main, 0.2);
    },

    get focused() {
      return setOpacity(this.main, 0.25);
    }

  },
  success: {
    main: geodeColors$1.green[40],
    dark: geodeColors$1.green[90],
    light: geodeColors$1.green[20],
    background: geodeColors$1.green[100],
    contrastText: geodeColors$1.black[100],

    get hover() {
      return setOpacity(this.main, 0.1);
    },

    get selected() {
      return setOpacity(this.main, 0.2);
    },

    get focused() {
      return setOpacity(this.main, 0.25);
    }

  },
  warning: {
    main: geodeColors$1.yellowOrange[30],
    dark: geodeColors$1.yellow[90],
    light: geodeColors$1.yellowOrange[10],
    background: geodeColors$1.yellow[100],
    contrastText: geodeColors$1.black[100],

    get hover() {
      return setOpacity(this.main, 0.1);
    },

    get selected() {
      return setOpacity(this.main, 0.2);
    },

    get focused() {
      return setOpacity(this.main, 0.25);
    }

  },
  error: {
    main: geodeColors$1.red[40],
    dark: geodeColors$1.red[90],
    light: geodeColors$1.red[10],
    background: geodeColors$1.red[100],
    contrastText: geodeColors$1.white[0],

    get hover() {
      return setOpacity(this.main, 0.1);
    },

    get selected() {
      return setOpacity(this.main, 0.2);
    },

    get focused() {
      return setOpacity(this.main, 0.25);
    }

  },
  miscellaneous: {
    backdrop: 'rgba(19, 23, 28, 0.5)',
    divider: 'rgba(137, 155, 168, 0.20)',
    filledInput: 'rgba(61, 73, 86, 0.4)',
    ratingFull: geodeColors$1.yellowOrange[20],
    ratingUnfilled: geodeColors$1.coolGray[60],
    snackBar: geodeColors$1.white[0],
    tooltip: 'rgba(216, 223, 228, 0.9)'
  },
  secondary: {
    contrastText: geodeColors$1.white[0],
    dark: geodeColors$1.violet[60],
    light: geodeColors$1.violet[10],
    main: geodeColors$1.violet[40],

    get hover() {
      return setOpacity(this.main, 0.1);
    },

    get selected() {
      return setOpacity(this.main, 0.2);
    },

    get focused() {
      return setOpacity(this.main, 0.25);
    }

  },
  text: {
    disabled: 'rgba(255, 255, 255, 0.25)',
    hint: 'rgba(255, 255, 255, 0.35)',
    primary: 'rgba(255, 255, 255, 0.9)',
    secondary: 'rgba(255, 255, 255, 0.6)'
  },
  stroke: {
    divider: 'rgba(255, 255, 255, 0.07)',
    stroke: 'rgba(255, 255, 255, 0.2)',
    hover: 'rgba(255, 255, 255, 0.5)',
    disabled: 'rgba(255, 255, 255, 0.07)'
  },
  type: 'dark'
};
const darkGeodePalette = {
  add: geodeColors$1.green[40],
  addRemove: geodeColors$1.red[50],
  appName01: geodeColors$1.blueViolet[50],
  backgroundPrimary: '#000000',
  backgroundQuaternary: geodeColors$1.coolGray[80],
  backgroundSecondary: geodeColors$1.coolGray[95],
  backgroundTertiary: geodeColors$1.coolGray[90],
  buttonDisabled: geodeColors$1.coolGray[95],
  buttonInactive: geodeColors$1.coolGray[80],
  cardBackgroundPrimary: geodeColors$1.gray[10],
  error: geodeColors$1.red[50],
  errorSuccess: geodeColors$1.green[40],
  geode: geodeColors$1.violet[50],
  labelPrimary: '#FFFFFF',
  labelQuaternary: geodeColors$1.coolGray[60],
  labelSecondary: geodeColors$1.coolGray[20],
  labelTertiary: geodeColors$1.coolGray[40],
  premiumWell: geodeColors$1.yellowOrange[20],
  priorityHigh: geodeColors$1.red[40],
  priorityLow: geodeColors$1.yellow[20],
  priorityMedium: geodeColors$1.orange[30],
  separator: geodeColors$1.coolGray[50],
  spreadsheetRow: '#000000',
  spreadsheetRowAlternate: geodeColors$1.coolGray[95],
  spreadsheetRowSelected: geodeColors$1.coolGray[70],
  streamBoe: '#FFFFFF',
  streamGas: geodeColors$1.red[40],
  streamNgl: geodeColors$1.blue[40],
  streamOil: geodeColors$1.green[30],
  streamWater: geodeColors$1.blue[20],
  tableviewBackground: '#000000',
  tableviewBackgroundSecondary: geodeColors$1.coolGray[95],
  tag01: geodeColors$1.blue[20],
  tag02: geodeColors$1.green[20],
  tag03: geodeColors$1.yellow[20],
  tag04: geodeColors$1.red[50],
  tag05: geodeColors$1.redViolet[40],
  tag06: geodeColors$1.blueGreen[30]
};
const p3DarkGeodePalette = {
  add: p3geodeColors.green[40],
  addRemove: p3geodeColors.red[50],
  appName01: p3geodeColors.blueViolet[50],
  backgroundPrimary: '#000000',
  backgroundQuaternary: p3geodeColors.coolGray[80],
  backgroundSecondary: p3geodeColors.coolGray[95],
  backgroundTertiary: p3geodeColors.coolGray[90],
  buttonDisabled: p3geodeColors.coolGray[95],
  buttonInactive: p3geodeColors.coolGray[80],
  cardBackgroundPrimary: p3geodeColors.gray[10],
  error: p3geodeColors.red[50],
  errorSuccess: p3geodeColors.green[40],
  geode: p3geodeColors.violet[50],
  labelPrimary: '#FFFFFF',
  labelQuaternary: p3geodeColors.coolGray[60],
  labelSecondary: p3geodeColors.coolGray[20],
  labelTertiary: p3geodeColors.coolGray[40],
  premiumWell: p3geodeColors.yellowOrange[20],
  priorityHigh: p3geodeColors.red[40],
  priorityLow: p3geodeColors.yellow[20],
  priorityMedium: p3geodeColors.orange[30],
  separator: p3geodeColors.coolGray[50],
  spreadsheetRow: '#000000',
  spreadsheetRowAlternate: p3geodeColors.coolGray[95],
  spreadsheetRowSelected: p3geodeColors.coolGray[70],
  streamBoe: '#FFFFFF',
  streamGas: p3geodeColors.red[40],
  streamNgl: p3geodeColors.blue[40],
  streamOil: p3geodeColors.green[30],
  streamWater: p3geodeColors.blue[20],
  tableviewBackground: '#000000',
  tableviewBackgroundSecondary: p3geodeColors.coolGray[95],
  tag01: p3geodeColors.blue[20],
  tag02: p3geodeColors.green[20],
  tag03: p3geodeColors.yellow[20],
  tag04: p3geodeColors.red[50],
  tag05: p3geodeColors.redViolet[40],
  tag06: p3geodeColors.blueGreen[30]
};
const lightGeodePalette = {
  add: geodeColors$1.green[40],
  addRemove: geodeColors$1.red[50],
  appName01: geodeColors$1.blueViolet[50],
  backgroundPrimary: '#FFFFFF',
  backgroundQuaternary: geodeColors$1.gray[30],
  backgroundSecondary: geodeColors$1.gray[0],
  backgroundTertiary: geodeColors$1.gray[10],
  buttonDisabled: geodeColors$1.gray[5],
  buttonInactive: geodeColors$1.gray[10],
  cardBackgroundPrimary: geodeColors$1.gray[10],
  error: geodeColors$1.red[50],
  errorSuccess: geodeColors$1.green[40],
  geode: geodeColors$1.violet[50],
  labelPrimary: '#000000',
  labelQuaternary: geodeColors$1.gray[10],
  labelSecondary: geodeColors$1.gray[50],
  labelTertiary: geodeColors$1.gray[30],
  premiumWell: geodeColors$1.yellowOrange[30],
  priorityHigh: geodeColors$1.red[50],
  priorityLow: geodeColors$1.yellow[40],
  priorityMedium: geodeColors$1.orange[30],
  separator: geodeColors$1.gray[40],
  spreadsheetRow: '#FFFFFF',
  spreadsheetRowAlternate: geodeColors$1.gray[0],
  spreadsheetRowSelected: geodeColors$1.coolGray[5],
  streamBoe: '#000000',
  streamGas: geodeColors$1.red[60],
  streamNgl: geodeColors$1.blue[60],
  streamOil: geodeColors$1.green[50],
  streamWater: geodeColors$1.blue[40],
  tableviewBackground: '#FFFFFF',
  tableviewBackgroundSecondary: geodeColors$1.gray[0],
  tag01: geodeColors$1.blue[40],
  tag02: geodeColors$1.green[40],
  tag03: geodeColors$1.yellowOrange[40],
  tag04: geodeColors$1.red[60],
  tag05: geodeColors$1.redViolet[50],
  tag06: geodeColors$1.blueGreen[40]
};
const p3LightGeodePalette = {
  add: p3geodeColors.green[40],
  addRemove: p3geodeColors.red[50],
  appName01: p3geodeColors.blueViolet[50],
  backgroundPrimary: '#FFFFFF',
  backgroundQuaternary: p3geodeColors.gray[30],
  backgroundSecondary: p3geodeColors.gray[0],
  backgroundTertiary: p3geodeColors.gray[10],
  buttonDisabled: p3geodeColors.gray[5],
  buttonInactive: p3geodeColors.gray[10],
  cardBackgroundPrimary: p3geodeColors.gray[10],
  error: p3geodeColors.red[50],
  errorSuccess: p3geodeColors.green[40],
  geode: p3geodeColors.violet[50],
  labelPrimary: '#000000',
  labelQuaternary: p3geodeColors.gray[10],
  labelSecondary: p3geodeColors.gray[50],
  labelTertiary: p3geodeColors.gray[30],
  premiumWell: p3geodeColors.yellowOrange[30],
  priorityHigh: p3geodeColors.red[50],
  priorityLow: p3geodeColors.yellow[40],
  priorityMedium: p3geodeColors.orange[30],
  separator: p3geodeColors.gray[40],
  spreadsheetRow: '#FFFFFF',
  spreadsheetRowAlternate: p3geodeColors.gray[0],
  spreadsheetRowSelected: p3geodeColors.coolGray[5],
  streamBoe: '#000000',
  streamGas: p3geodeColors.red[60],
  streamNgl: p3geodeColors.blue[60],
  streamOil: p3geodeColors.green[50],
  streamWater: p3geodeColors.blue[40],
  tableviewBackground: '#FFFFFF',
  tableviewBackgroundSecondary: p3geodeColors.gray[0],
  tag01: p3geodeColors.blue[40],
  tag02: p3geodeColors.green[40],
  tag03: p3geodeColors.yellowOrange[40],
  tag04: p3geodeColors.red[60],
  tag05: p3geodeColors.redViolet[50],
  tag06: p3geodeColors.blueGreen[40]
};

const lightOverrides = {
  MuiCssBaseline: {
    '@global': {
      '* > .MuiCardHeader-action': {
        margin: 0
      },
      '* > code': {
        backgroundColor: geodeColors$1.gray[20],
        padding: '4px'
      },
      a: {
        color: geodeColors$1.blue[40],
        textDecoration: 'none'
      },
      'a:hover': {
        textDecoration: 'underline'
      },
      'a:visited': {
        color: geodeColors$1.violet[50]
      }
    }
  },
  MuiListItemIcon: {
    root: {
      color: 'inherit'
    }
  }
};
const darkOverrides = {
  MuiCssBaseline: {
    '@global': {
      '* > .MuiCardHeader-action': {
        margin: 0
      },
      '* > code': {
        backgroundColor: geodeColors$1.gray[80],
        padding: '4px'
      },
      a: {
        color: geodeColors$1.blue[40],
        textDecoration: 'none'
      },
      'a:hover': {
        textDecoration: 'underline'
      },
      'a:visited': {
        color: geodeColors$1.violet[40]
      }
    }
  },
  MuiListItemIcon: {
    root: {
      color: 'inherit'
    }
  }
};

const shadows = ['none', '0px 2px 1px 0px rgba(0, 0, 0, 0.1),0px 1px 1px 0px rgba(0, 0, 0, 0.07),0px 1px 3px 0px rgba(0, 0, 0, 0.06)', '0px 3px 2px 0px rgba(0, 0, 0, 0.1),0px 2px 2px 0px rgba(0, 0, 0, 0.07),0px 1px 5px 0px rgba(0, 0, 0, 0.06)', '0px 3px 3px 0px rgba(0, 0, 0, 0.1),0px 3px 4px 0px rgba(0, 0, 0, 0.07),0px 1px 8px 0px rgba(0, 0, 0, 0.06)', '0px 2px 4px 0px rgba(0, 0, 0, 0.1),0px 4px 5px 0px rgba(0, 0, 0, 0.07),0px 1px 10px 0px rgba(0, 0, 0, 0.06)', '0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12)', '0px 3px 5px 0px rgba(0, 0, 0, 0.1),0px 6px 10px 0px rgba(0, 0, 0, 0.07),0px 1px 18px 0px rgba(0, 0, 0, 0.06)', '0px 4px 5px -2px rgba(0,0,0,0.2),0px 7px 10px 1px rgba(0,0,0,0.14),0px 2px 16px 1px rgba(0,0,0,0.12)', '0px 5px 5px 0px rgba(0, 0, 0, 0.1),0px 8px 10px 0px rgba(0, 0, 0, 0.07),0px 3px 14px 0px rgba(0, 0, 0, 0.06)', '0px 9px 12px 1px rgba(0, 0, 0, 0.14),0px 3px 16px 2px rgba(0, 0, 0, 0.12),0px 5px 6px -3px rgba(0, 0, 0, 0.2)', '0px 6px 6px -3px rgba(0,0,0,0.2),0px 10px 14px 1px rgba(0,0,0,0.14),0px 4px 18px 3px rgba(0,0,0,0.12)', '0px 6px 7px -4px rgba(0,0,0,0.2),0px 11px 15px 1px rgba(0,0,0,0.14),0px 4px 20px 3px rgba(0,0,0,0.12)', '0px 7px 8px 0px rgba(0, 0, 0, 0.1),0px 12px 17px 0px rgba(0, 0, 0, 0.07),0px 5px 22px 0px rgba(0, 0, 0, 0.06)', '0px 7px 8px -4px rgba(0,0,0,0.2),0px 13px 19px 2px rgba(0,0,0,0.14),0px 5px 24px 4px rgba(0,0,0,0.12)', '0px 7px 9px -4px rgba(0,0,0,0.2),0px 14px 21px 2px rgba(0,0,0,0.14),0px 5px 26px 4px rgba(0,0,0,0.12)', '0px 8px 9px -5px rgba(0,0,0,0.2),0px 15px 22px 2px rgba(0,0,0,0.14),0px 6px 28px 5px rgba(0,0,0,0.12)', '0px 8px 10px 0px rgba(0, 0, 0, 0.1),0px 16px 24px 0px rgba(0, 0, 0, 0.07),0px 6px 30px 0px rgba(0, 0, 0, 0.06)', '0px 8px 11px -5px rgba(0,0,0,0.2),0px 17px 26px 2px rgba(0,0,0,0.14),0px 6px 32px 5px rgba(0,0,0,0.12)', '0px 9px 11px -5px rgba(0,0,0,0.2),0px 18px 28px 2px rgba(0,0,0,0.14),0px 7px 34px 6px rgba(0,0,0,0.12)', '0px 9px 12px -6px rgba(0,0,0,0.2),0px 19px 29px 2px rgba(0,0,0,0.14),0px 7px 36px 6px rgba(0,0,0,0.12)', '0px 10px 13px -6px rgba(0,0,0,0.2),0px 20px 31px 3px rgba(0,0,0,0.14),0px 8px 38px 7px rgba(0,0,0,0.12)', '0px 10px 13px -6px rgba(0,0,0,0.2),0px 21px 33px 3px rgba(0,0,0,0.14),0px 8px 40px 7px rgba(0,0,0,0.12)', '0px 10px 14px -6px rgba(0,0,0,0.2),0px 22px 35px 3px rgba(0,0,0,0.14),0px 8px 42px 7px rgba(0,0,0,0.12)', '0px 11px 14px -7px rgba(0,0,0,0.2),0px 23px 36px 3px rgba(0,0,0,0.14),0px 9px 44px 8px rgba(0,0,0,0.12)', '0px 11px 15px 0px rgba(0, 0, 0, 0.1),0px 24px 38px 0px rgba(0, 0, 0, 0.07),0px 9px 46px 0px rgba(0, 0, 0, 0.06)'];

const typography = {
  h1: {
    fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
    fontWeight: 300,
    fontSize: '96px',
    lineHeight: '112px',
    letterSpacing: '-1.5px'
  },
  h2: {
    fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
    fontWeight: 300,
    fontSize: '60px',
    lineHeight: '72px',
    letterSpacing: '-0.5px'
  },
  h3: {
    fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
    fontWeight: 400,
    fontSize: '48px',
    lineHeight: '56px',
    letterSpacing: '0px'
  },
  h4: {
    fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
    fontWeight: 400,
    fontSize: '34px',
    lineHeight: '42px',
    letterSpacing: '.25px'
  },
  h5: {
    fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
    fontWeight: 400,
    fontSize: '24px',
    lineHeight: '32px',
    letterSpacing: '0px'
  },
  h6: {
    fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
    fontWeight: 500,
    fontSize: '20px',
    lineHeight: '32px',
    letterSpacing: '.15px'
  },
  subtitle1: {
    fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
    fontWeight: 400,
    fontSize: '16px',
    lineHeight: '28px',
    letterSpacing: '.15px'
  },
  subtitle2: {
    fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
    fontWeight: 500,
    fontSize: '14px',
    lineHeight: '22px',
    letterSpacing: '.1px'
  },
  body1: {
    fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
    fontWeight: 400,
    fontSize: '16px',
    lineHeight: '24px',
    letterSpacing: '.15px'
  },
  body2: {
    fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
    fontWeight: 400,
    fontSize: '14px',
    lineHeight: '20px',
    letterSpacing: '.15px'
  },
  caption: {
    fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
    fontWeight: 400,
    fontSize: '12px',
    lineHeight: '20px',
    letterSpacing: '.4px'
  },
  overline: {
    fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
    fontWeight: 400,
    fontSize: '12px',
    lineHeight: '20px',
    letterSpacing: '1px'
  }
};

const p3check = window && window.CSS && window.CSS.supports && window.CSS.supports('color', 'color(display-p3 1 1 1)');
const supportsP3 = () => {
  return p3check ? {
    geodeColors: p3geodeColors,
    darkGeodePalette: p3DarkGeodePalette,
    lightGeodePalette: p3LightGeodePalette
  } : {
    geodeColors: geodeColors$1,
    darkGeodePalette,
    lightGeodePalette
  };
};

const {
  geodeColors
} = supportsP3(); // colormap

const geodeDarken = (startingColor, steps) => {
  const newColors = Object.keys(geodeColors).map(key => {
    const color = geodeColors[key]; // The colorkeys add `05` as a usability entry.

    const colorKeys = Object.keys(color).filter(k => k !== '05');
    const existingGeodeColorObject = colorKeys.find(entry => color[entry] === startingColor);
    if (!existingGeodeColorObject) return undefined;
    const index = colorKeys.indexOf(existingGeodeColorObject);
    const newIndex = index + steps;

    if (newIndex > colorKeys.length) {
      const lastColorEntry = colorKeys[colorKeys.length - 1];
      return color[lastColorEntry];
    }

    if (newIndex < 0) {
      return color[0];
    }

    const newColorEntry = colorKeys[newIndex];
    return color[newColorEntry];
  });
  return newColors.find(r => r) || startingColor;
};
const geodeLighten = (startingColor, steps) => geodeDarken(startingColor, steps * -1);

export { darkGeodePalette, darkOverrides, darkPalette, geodeColors$1 as geodeColors, geodeDarken, geodeLighten, lightGeodePalette, lightOverrides, lightPalette, p3DarkGeodePalette, p3LightGeodePalette, p3geodeColors, setOpacity, shadows, supportsP3, typography };
